webapp/src/core.ts

```diff
@@ -154,6 +155,7 @@ export interface DialogOptions {
     header: string;
     body?: string;
     jsx?: JSX.Element;
+    jsxd?: () => JSX.Element; // dynamic-er version of jsx
```

Comment:
how required is this / how much work would it be to just implement it as jsx? Looks like it's just used in a few dialogues, and it seems like this is a part that's fighting with react a bit and might make it harder to reason about (mainly if it's used more than just one or two places)

Reply:
The problem with JSX is that it is completely static. So it is not possible to have some local state for the dialog (for example needed to track which radio button is pressed).
Our entire dialog scheme is fighting react.

Reply:
Yeah, that makes sense to me. I guess an option would be to make dialogs classes, but that seems like a decent chunk of work that I wouldn't want to see added to a PR that's already this large hah

Reply:
To be clear, this is already an improvement compared to using the DOM element created by React in some dialogs.